pkgs/data/icons/candy-icons/default.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs }:
+
+pkgs.stdenv.mkDerivation {
+
+  pname = "candy-icons";
+  version = "0.6.8";
+
+  nativeBuildInputs = [ pkgs.gtk3 ];
+
+  src = pkgs.fetchurl {
+    url = "https://github.com/EliverLara/candy-icons/archive/37b4566b4bfcdad99131a236ef1fc74db89a302e.zip";
+    sha256 = "fgey2MduS9alUKiFhZr6KTKhYa68zh6LzEIcg/amrOo=";
+  };
+
+  dontUnpack = true;
+
+  installPhase = ''
+    mkdir -p $out/share/icons
+    ${pkgs.unzip}/bin/unzip $src -d $out/share/icons/
+    mv $out/share/icons/candy-icons-* $out/share/icons/candy-icons
+
+    for theme in $out/share/icons/*; do
+      gtk-update-icon-cache --force $theme
+    done
+  '';
+}
```

Comment:
This is going to need a `meta` block: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes.

I think you should also create a commit that adds yourself to the maintainer list, so you can be marked as package maintainer. Have a look at [some of the other closed first-time contribution prs](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+label%3A%2212.+first-time+contribution%22+is%3Aclosed+init) to see how this might look like.